### PR TITLE
Spacing control: Replace sides dropdwon with link button

### LINKED ISSUE #63963: Make SpacingSizesControl more intuitive
The current experience of editing a specific sides of the SpacingSizesControl is difficult to parse. 

Here's why: 

1. The "controlName options" icon (the `DropdownMenu` that lists the individual sides) has a very similar icon to the icons to the left of the RangeControls for each. It's not clear what this icon does, and because it's so similar it's easy to mistake the other icons for this (and vise versa).  
2. We're prioritizing applying one side at a time, over _any_ side at once. 
3. If you switch to any of the individual sides, the UI will **only** show that current side. You can have a value applied to others, but the UI only shows the current side. 
4. If you switch to any of the individual sides and want to change another, you have to switch back to another state to change another. The alternative is to always change to the "Custom" option, which shows all sides. 
5. We have the notion of "Link sides" and "Unlink sides" settings, as seen in the border and border radius controls. Why should SpacingSizesControl perform differently? It's the only control with an additional `DropdownMenu` to switch views within the component. 

The original effort to only show one side at a time was an effort to save vertical space in the Inspector, but with the current improvements to this UI, I think we can afford to use this space—to make it much more intuitive to use. 

### Current

https://github.com/user-attachments/assets/6660b6c8-f4d9-41e2-94d5-963d94ba1bf5

### Proposed

You select the "Unlink sides" icon and the sides are unlinked. No DropdownMenu to invoke singular sides. I've also updated icons to make them easier to scan and follow a vertical line.

https://github.com/user-attachments/assets/08b01fc4-809b-4803-a3bf-279dc3e1c758

[Here's a prototype](https://www.figma.com/proto/oEm1ck3f5gfGYyOR1Fs71T/Gutenberg-Ideas?page-id=2718%3A4350&node-id=2727-2141&viewport=-2826%2C6435%2C2.03&t=RGhGZEliPmJf678u-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=2727%3A2141)

### PR DESCRIPTION
**Pull request by madhusudhand**
## What?

This changes removes sides dropdown from the `SpacingSizesControl` (Padding and Margin), and replaces it with a link button.

Fixed #63963 (partially)

Note: no updates to icons in this iteration.

## Testing Instructions

* Choose a block such as gallery (with spacing settings enabled) and observe that sides dropdown is now replaced with link button.
* Link should toggle between `horizontal & vertical` view and `all sides` custom view.
* It should be the same behaviour in the global style panel

## Screenshots or screencast 

image
image

### CODE REVIEW

File: packages/block-editor/src/components/spacing-sizes-control/linked-button.js
Comment (vcanales) [inline packages/block-editor/src/components/spacing-sizes-control/linked-button.js:12]: I think there might be room for improvement in the wording of this label:

- Adding "sides" to clarify what we are linking: "Link sides"/"Unlink sides"

Also, alternative wordings:
- "All sides"/"Individual sides"
- "Uniform sides"/"Custom sides"

I like "Uniform sides"/"Custom sides" because it helps me form a mental image of what might happen when I activate the linked mode.

cc. @richtabor

**Review by vcanales — COMMENTED** [review]
Code-wise LGTM, and it works as advertised;  thank you! 
I left a comment regarding the wording of the label.

File: packages/block-editor/src/components/spacing-sizes-control/linked-button.js
Comment (richtabor) [reply packages/block-editor/src/components/spacing-sizes-control/linked-button.js:12]: I think "Unlink padding" and "Link padding" is ok (lowercase). 

It's not always the case that padding and margin are applied on all sides—generally horizontal and vertical—so uniform doesn't scale well. We could potentially adjust in the future though.

File: packages/block-editor/src/components/spacing-sizes-control/linked-button.js
Comment (richtabor) [inline packages/block-editor/src/components/spacing-sizes-control/linked-button.js]: suppose this class is not needed, yea?

**Review by richtabor — COMMENTED** [review]
Feels much simpler. Like it. 

One tiny detail: let's make the "Margin" and "Padding" labels in the tooltip lowercase, to match others.

File: packages/block-editor/src/components/spacing-sizes-control/linked-button.js
Comment (madhusudhand) [reply packages/block-editor/src/components/spacing-sizes-control/linked-button.js:12]: fixed in eb85d052955c929051ae36d00c39c13459e0a13b

File: packages/block-editor/src/components/spacing-sizes-control/linked-button.js
Comment (madhusudhand) [reply packages/block-editor/src/components/spacing-sizes-control/linked-button.js]: Yea. removed here eb85d052955c929051ae36d00c39c13459e0a13b

**Review by mikachan — APPROVED** [review]
This is working well for me 🎉 

- The controls still work as expected in my testing ✅ 
- I can see "padding" and "margin" are both lowercase in the button tooltip ✅

**Comment by t-hamano** [comment]
This PR appears to have made the block spacing UI unavailable if **the theme defines a default gap for blocks**.

For example, the TT4 theme defines [a default block gap for the Buttons block](https://github.com/WordPress/wordpress-develop/blob/be75ea8e73a6d64616fea64e9ccf76ac1268a61e/src/wp-content/themes/twentytwentyfour/theme.json#L324).

If you access the global styles for the Buttons block, you’ll see something like this:

image

Before this PR was merged, it looked like this:

image

However, this is not ideal, and since the Buttons block [supports the axial gap](https://github.com/WordPress/gutenberg/blob/1df91fb8289db22837ab68030ef7d87c90a2ed72/packages/block-library/src/buttons/block.json#L24), it should look like this:

image

This issue has also been reported here: https://github.com/WordPress/gutenberg/pull/64971#pullrequestreview-2295600451

My guess is that it has to do with updating the `getInitialView()` function.

**Comment by madhusudhand** [comment]
@t-hamano Your are right. It is because the function `getInitialView` returns custom view and block spacing control doesn't render any custom control because it supports only `horizontal` and `vertical` not all sides.

In #65371 the function is restored to its original code and included additional condition and it should now be fixed.

**Comment by mcsf** [comment]
I stumbled upon this PR after spotting an incorrect translation submitted via GlotPress for my main locale (pt_PT).

The way that strings are built in this PR (as well as preceding strings that the PR is built atop) lends itself to translation mistakes for a few reasons. The term _link_ is almost always used in the hypertextual sense, so translators will struggle to understand `Link %1$s`, and I believe the translator note ("1. Type of spacing being modified (padding, margin, etc).") is not clear enough.

We also need syntagmatic information to understand the "link" lexeme: is it a noun? a verb? infinitive or imperative? Few languages are as weakly inflected as English, and using `_x` to specify that function — in addition to a translation note — would go a long way.

Furthermore, as always, stringing together strings with variables works in English but lends itself to fragile translations in many languages. I'm referring to the fact that we call `sprintf( __( '%s options' ), thing )` or `sprintf( __( 'Link %s' ), thing )` without acknowledging that the fragment `thing` may need to be adequately inflected in a given locale. [See a complete example I provided for Finnish and Romance languages here.](https://github.com/WordPress/gutenberg/issues/48963#issuecomment-1570661831) (WordPress post type labels [follow the same logic](https://developer.wordpress.org/reference/functions/get_post_type_labels/).) This is not a deal breaker, because we still choose to interpolate across our UI when the tradeoff seems acceptable, but note that if we know beforehand what values could be fed into the interpolation then we should probably use static strings that allow translators to produce better translations:

Don't | Do
-|-
`'%s options'`, `'margin'`, `'padding'` | `'margin options'`, `'padding options'`

The above may or may not apply to this PR, I'll leave that up to you, since it depends on how generic and reusable you want some of the components to be.

Finally, but most importantly: is "Link padding" even the right string for the UI in question? That's not how I would describe that button's effect. It's more along the lines of "Link sides", "Link directions", "Link properties", etc. Actually, maybe not so much "link" as "group". My point here is that I suspect that we can come up with a better label that conveniently also does _not_ depend on the property being manipulated.

@madhusudhand @mikachan let me know if you need my help.

**Comment by t-hamano** [comment]
@mcsf Thanks for the feedback.

> Finally, but most importantly: is "Link padding" even the right string for the UI in question? That's not how I would describe that button's effect. It's more along the lines of "Link sides", "Link directions", "Link properties", etc. Actually, maybe not so much "link" as "group". My point here is that I suspect that we can come up with a better label that conveniently also does not depend on the property being manipulated.

I agree with you.

`SpacingSizesControl` is a public component that takes a `label` as a prop, and this label is applied to the `Link %s / Unlink %s` placeholder. Because the label passed as a property is dynamic, it would certainly not be appropriate to use this label as part of the linked button text.

I prefer a hard-coded label of `Link sides / Unlink sides`, which [has been used in the past](https://github.com/WordPress/gutenberg/blob/4c60efe0d991c202b6bdc2f93d56e4a8273ab074/packages/block-editor/src/components/spacing-sizes-control/linked-button.js#L9), and the `BorderRadiusControl` component [has a similar hard-coded label](https://github.com/WordPress/gutenberg/blob/f83c59253907969de16e172eb5a0fc7bc1f90f9c/packages/block-editor/src/components/border-radius-control/linked-button.js#L9).

**Pull request merged**
